src/playlist.js

```diff
@@ -216,21 +216,28 @@ export const sumDurations = function(playlist, startIndex, endIndex) {
  * window which is the duration of the last segment plus 2 target durations from the end
  * of the playlist.
  *
+ * A liveEdgePadding can be provided which will be used instead of calculating the safe live edge.
+ * This corresponds to suggestedPresentationDelay in DASH manifests.
+ *
  * @param {Object} playlist
  *        a media playlist object
+ * @param {Number} liveEdgePadding
+ *        A number in seconds indicating how far from the end we want to be.
+ *        If provided, this value is used instead of calculating the safe live index from the target durations.
+ *        Corresponds to suggestedPresentationDelay in DASH manifests.
  * @return {Number}
  *         The media index of the segment at the safe live point. 0 if there is no "safe"
  *         point.
  * @function safeLiveIndex
  */
-export const safeLiveIndex = function(playlist) {
+export const safeLiveIndex = function(playlist, liveEdgePadding) {
   if (!playlist.segments.length) {
     return 0;
   }
 
   let i = playlist.segments.length - 1;
   let distanceFromEnd = playlist.segments[i].duration || playlist.targetDuration;
-  const safeDistance = distanceFromEnd + playlist.targetDuration * 2;
+  const safeDistance = liveEdgePadding || (distanceFromEnd + playlist.targetDuration * 2);
```

Comment:
What if someone wants to explicitly put `0` in the MPD for suggestedPresentationDelay? (IOP recommends it be at least `minBufferTime`, but nothing in spec prevents it from being set to `0`)

Reply:
Yup, that should be valid. I guess that's why having some tests added would be good.